### ISSUE TITLE
Check `file.managed` replacement need

### ISSUE THREAD

**Issue by mjtrangoni** [warpnet/salt-lint#136]
Hi,

I have one case for [file.managed](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.managed), where salt self changes the default value of *replace* from True to False, printing this warning
```bash
[WARNING ] State for file: /boot/efi/EFI/centos/grub.cfg - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
``` 
We are changing here only the file mode and ownership so there is no need of replacing the file.

problematic state:

```yaml
# EFI mode 0700
{% for grub_cfg in ['/boot/efi/EFI/centos/grub.cfg', '/boot/efi/EFI/redhat/grub.cfg'] %}
cis_grub.cfg_{{ grub_cfg }}:
  file.managed:
    - name: {{ grub_cfg }}
    - user: root
    - group: root
    - mode: '0700'
    - onlyif: test -f {{ grub_cfg }}
{% endfor %}
```

fixed state:

```yaml
# EFI mode 0700
{% for grub_cfg in ['/boot/efi/EFI/centos/grub.cfg', '/boot/efi/EFI/redhat/grub.cfg'] %}
cis_grub.cfg_{{ grub_cfg }}:
  file.managed:
    - name: {{ grub_cfg }}
    - user: root
    - group: root
    - mode: '0700'
    - replace: False
    - onlyif: test -f {{ grub_cfg }}
{% endfor %}
```

**Comment by roaldnefs** [comment]
Thanks @mjtrangoni for your feature request!

In order to implement this rule we could render the states as YAML and just check if  'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined.
